stdlib/LinearAlgebra/src/tridiag.jl

```diff
@@ -161,10 +161,11 @@ function mul!(C::StridedVecOrMat, S::SymTridiagonal, B::StridedVecOrMat)
     β = S.ev
     @inbounds begin
         for j = 1:n
-            x₀, x₊ = B[1, j], B[2, j]
-            β₀ = β[1]
-            C[1, j] = α[1]*x₀ + x₊*β₀
-            for i = 2:m - 1
+            x₊ = B[1, j]
+            x₀ = zero(x₊)
+            # If n=1 then β[1] is out of bounds
+            β₀ = n > 1 ? zero(β[1]) : zero(eltype(β))
```

Comment:
Shouldn't the condition be on `m` rather than `n`?

Apart from that, it might worthwhile avoiding the need for calling `zero` altogether by completely special-casing `m==1`.

Reply:
> Shouldn't the condition be on m rather than n?

Indeed it should.

> Apart from that, it might worthwhile avoiding the need for calling zero altogether by completely special-casing m==1.

The variables would need to be initialized anyway so I prefer not to special case the store for `m==1`.

Reply:
My thinking was
```julia
@inbounds begin
    if m > 1
        for j = 1:n
            # old code
        end
    else
        for j = 1:n
            x₊ = B[1, j]
            C[1, j] = α[1]*x₊
        end
    end
end
```
(where the `else` part could be simplified to a broadcast, of course).